src/main.rs

```diff
@@ -0,0 +1,77 @@
+mod dockerfile;
+
+use dockerfile::{Dockerfile, ValidationResult};
+use glob::glob;
+use std::{error::Error, process};
+
+fn main() {
+    let dockerfiles = get_dockerfiles().expect("[ERROR] Failed to get Dockerfiles.");
+    if dockerfiles.is_empty() {
+        skip();
+    };
+
+    let invalid = validate_dockerfiles(&dockerfiles);
+    if invalid.is_empty() {
+        pass();
+    }
+
+    fail(&invalid);
+}
+
+fn get_dockerfiles() -> Result<Vec<Dockerfile>, Box<dyn Error>> {
+    glob("**/*Dockerfile*")?
+        .map(|path| Dockerfile::new(path?))
+        .collect()
+}
+
+fn validate_dockerfiles(dockerfiles: &[Dockerfile]) -> Vec<(&Dockerfile, ValidationResult)> {
+    dockerfiles
+        .iter()
+        .map(|dockerfile| (dockerfile, dockerfile.validate()))
+        .filter(|result| !result.1.invalid_images.is_empty())
+        .collect()
+}
+
+fn skip() {
+    println!("[PASS] No Dockerfiles found.");
```

Comment:
I know this is very simple and doesn't require a "logging library" per se but getting used to using `log` for these cases is better practice imo: https://docs.rs/log/0.4.10/log/

Reply:
will give this a try 💯

Reply:
which logger implementation do you recommend for executables?

Reply:
From the docs on this one:
`The executable itself may use the log crate to log as well.`

So could be a winner just to use this.

Reply:
strange. gave that a try, but it didn't seem to output anything 😞

Reply:
strange. gave that a try but it didn't seem to output any logs to stdout 😞

Reply:
strange.. gave that a try but it didn't seem to output any logs to stdout 😞